pre_commit_hooks/requirements_txt_fixer.py

```diff
@@ -61,6 +61,9 @@ def fix_requirements(f):
         elif line.startswith(b'#') or line.strip() == b'':
             requirement.comments.append(line)
         else:
+            # Automatically fix 'package=version' to 'package==version'
+            if b'=' in line and b'==' not in line and b'egg=' not in line:
```

Comment:
it looks like there's a few other cases that aren't covered here.  Reading through https://pip.pypa.io/en/stable/reference/pip_install/#requirement-specifiers these are things that might be valid in a `requirements.txt`:

```
mypy;python_version>="3.5"
foopkg --hash=sha256:12341234123412341234
--index-url=https://example.com/simple
```

(it would be good to add a test for these as well!)

Reply:
Yeah, I came to the same conclusion (and created #331).